app/lib/account/backend.dart

```diff
@@ -88,16 +86,27 @@ UserSessionData? get userSessionData =>
 /// When no associated User entry exists in Datastore, this method will create
 /// a new one. When the authenticated email of the user changes, the email
 /// field will be updated to the latest one.
-Future<User> requireAuthenticatedUser({AuthSource? source}) async {
+Future<User> requireAuthenticatedUser({String? expectedAudience}) async {
```

Comment:
Let's make it:
```dart
Future<AuthenticatedAgent> requireAuthenticatedAgent({Set<String>? authorizedAudiences}) async {
  authorizedAudiences ??= [
     // TODO: List of all audiences that may be allowed.
  ];
}

class AuthenticatedUser extends AuthenticatedAgent {
  final User user;
  final String audience; // TODO: Let's add this field
}

Future<AuthenticatedAgent> requireAuthenticatedClient() {
  return await requireAuthenticatedAgent(authorizedAudiences: [
    'https://pub.dev',
    // TODO: Audience from client with oauth
  ]);
}
```

Reply:
I was planning to have something like this as the next step, to limit the number of moving parts in a single change.

Reply:
okay, that's reasonable :D